include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -8945,6 +8945,23 @@ def err_bounds_type_annotation_lost_checking : Error<
 
   def err_checked_scope_no_variadic_func_for_expression : Error<
     "cannot use a variable arguments function in a checked scope or function">;
+
+  def err_bounds_cast_array_pointer_type_with_single_syntax
+      : Error<"invalid bounds casting : cast array pointer type with single "
+              "bounds casting syntax :  %0">;
+
+  def err_bounds_cast_integral_with_single_syntax
+      : Error<"invalid bounds casting : cast integral type with single bounds "
+              "casting syntax :  %0">;
+
+  def err_bounds_cast_single_pointer_type_with_array_syntax
+      : Error<"invalid bounds casting : cast single pointer type with array "
```

Comment:
I would simplify this error message to be: `invalid bounds cast: expected Array_ptr type`.  The error message should point directly at the problem target type.